cp23/hpi-fun2/motivation/problem.tex

```diff
@@ -0,0 +1,56 @@
+\input{template.tex}
+
+\begin{document}
+
+\makeheader
+
+One day, Christian, the CEO of \textit{Super Arrangement Productions}, entered one of their offices
+to talk to an employee who complained about the lack of motivation in their company. 
+Christian was quite surprised, that not only the department, but the whole company was not motivated
+at all.
+
+That is why Christian made the decision to hire a motivation agency to send the best of their coaches
+to help motivate the people again.
+
+In a brief knowledge exchange with the head of the coaches, Christian learned the following things:
+
+\begin{itemize}
+    \item Each employee has a demotivational level, which tells how much effort it needs for a coach to motivate the employee again.
+    \item The agency sends their best coaches with each having a specific motivational level.
+    \item Motivating employees is exhausting, which means that coaches and employees motivating other employees will be less motivated, equally to the level of demotivation.
+    \item Employees are motivated by at least one person, but an employee will not be willing to get motivated by a person that higher hierarchy layers did not get motivated by.
+    \item After being motivated again, employees are willing to motivate other employees, but only if those employees are directly or indirectly under their current position.
+\end{itemize}
+
+Since Christian enforced saving measures, he wants to spend as little as possible. Therefore, as few coaches
+as possible should be sent. They will start working with Christian himself.
+
+Even though the whole company should become motivated again, some people simply might not be motivatable.
+Christian wants to know who these people are.
+
+\paragraph*{Input}
+
+The input consists of three lines, describing a single test case.
+The first line contains an integer $n$ ($1 \leq n \leq 5 \cdot 10^4$), the number of employees and the motivational level $l$ of the coaches ($1 \leq l \leq 10**5$).
+The second line contains $n$ integers. The $i$th integer is $m_i$ ($-10**6 \leq m_i \leq 10**6$), the motivational level of employee $i$.
+The third line contains $n - 1$ integers. The $i$th integer is $bi+1$ ($1 \leq b_{i+1} < i + 1$), the direct boss of colleague $i + 1$.
```

Comment:
Just out of curiosity ^^. Why did you chase the parent representation over the edge list representation?

Reply:
I feel like it fits better the way the problem is described, with the company hierarchy 😄 There was some task, which had a similar problem setup (I believe it was `party`) and I copied the input from there.